src/units/abilities.cpp

```diff
@@ -448,15 +483,18 @@ std::pair<int,map_location> unit_ability_list::lowest(const std::string& key, in
 	int stack = 0;
 	for (unit_ability const &p : cfgs_)
 	{
-		int value = (*p.first)[key].to_int(def);
+		int value = get_single_ability_value((*p.first)[key], def, p.second, loc(),[&](const wfl::formula& formula, wfl::map_formula_callable& callable) {
+			return formula.evaluate(callable).as_int();
+		});
+
 		if ((*p.first)["cumulative"].to_bool()) {
 			stack += value;
 			if (value < 0) value = -value;
-			if (only_cumulative && value <= abs_max) {
+			if (only_cumulative && !comp(value, abs_max)) {
 				abs_max = value;
 				best_loc = p.second;
 			}
-		} else if (only_cumulative || value < flat) {
+		} else if (only_cumulative || comp(flat, value)) {
```

Comment:
Ah, alright then. I think I'd recommend changing the name though, since it's no longer a "highest" or "lowest". Maybe something like "get_extremum"? Since "extremum" is the generalization of "minimum" and "maximum".